src/FlaUILibrary/flaui/module/element.py

```diff
@@ -221,7 +221,10 @@ def _get_element_by_xpath(self, xpath: str):
         Args:
             xpath (string): XPath identifier from element.
         """
-        return self._automation.GetDesktop().FindFirstByXPath(xpath)
+        try:
+            return self._automation.GetDesktop().FindFirstByXPath(xpath)
+        except ElementNotAvailableException:
+            return None

```

Comment:
It won't, because if you remember this function wasn't there at all, we have only added for the stable clicks, this is why also only references of this function is in stable clicks.

Reply:
So in this case the follow up code requires either an element or null.

Reply:
The design from this method is quite easy. It should never return None because this is an centralized method to get an element object. For your use case it's not important to return None but if this happens on any other keyword it will return None and undefined behavior. So this is not a clean solution becaus this is an logic change.

All methods must check now if this value is None and react on it. This is not the case. It should be guarantee that you can identify an issue here. So afterwards for my opinion you should handle this type of exception not here.

Reply:
This is not a centralized method which will be used from other Classes, this is a private function. Exceptionally used by stable click because in stable click where this function called needed to be abstracted from the general timeout of flauilibrary. But whatever, what is your suggestion to let the function fail?
So i need to catch it in the mouse function the fails and let it pass. I can also do it that way.

Reply:
I have looked at the central processing chain.

The UIA interface has a method GetElement(...) --> https://github.com/GDATASoftwareAG/robotframework-flaui/blob/main/src/FlaUILibrary/flaui/automation/uia.py#L72

This is used globally for all keywords that require elements for further interaction --> Example --> https://github.com/GDATASoftwareAG/robotframework-flaui/blob/main/src/FlaUILibrary/keywords/grid.py#L46
 
However, _get_element is included with an if to check for non-null. If null xpath not found is thrown --> https://github.com/GDATASoftwareAG/robotframework-flaui/blob/main/src/FlaUILibrary/flaui/module/element.py#L204

Okay so far this looks good.

Translated with DeepL.com (free version)